spec/lib/jobs/generate-tag-spec.js

```diff
@@ -0,0 +1,243 @@
+                    path: 'dmi.dmi.system.manufacturer',
+                    equals: 'Renasar'
+                }
+            ]
+        }, {
+            id: '2',
+            createdAt: new Date('Feb 01 2015 12:00:00'),
+            name: 'Test Tag 2',
+            rules: [
+                {
+                    path: 'dmi.dmi.system.manufacturer',
+                    equals: 'Renasar'
+                },
+                {
+                    path: 'dmi.dmi.system.product_name',
+                    equals: '1A'
+                }
+            ]
+        }];
+        waterline.tags.find.resolves(tags);
+        waterline.catalogs.findMostRecent.resolves(catalog1);
+
+        return job.run()
+        .then(function() {
+            expect(waterline.nodes.addTags).to.have.been.calledOnce;
+            expect(waterline.nodes.addTags).to.have.been.calledWith(['Test Tag', 'Test Tag 2']);
+        });
+    });
+
+    it('assigns a matching tag against multiple catalogs', function() {
+        var job = new GenerateTag({}, { target: 'bc7dab7e8fb7d6abf8e7d6ab' }, uuid.v4());
+        var tag = {
+            id: '1',
+            createdAt: new Date('Feb 01 2015 12:00:00'),
+            name: 'Test Tag',
+            rules: [
+                {
+                    path: 'dmi.dmi.system.product_name',
+                    equals: '1A'
+                },
+                {
+                    path: 'bmc.IP Address',
+                    regex: /^10\.[0-9]+.[0-9]+.[0-9]+$/
+                }
+            ]
+        };
+        waterline.tags.find.resolves([tag]);
+        waterline.catalogs.findMostRecent.onCall(0).resolves(catalog1);
+        waterline.catalogs.findMostRecent.onCall(1).resolves(catalog2);
+
+        return job.run()
+        .then(function() {
+            expect(waterline.nodes.addTags).to.have.been.calledOnce;
+            expect(waterline.nodes.addTags).to.have.been.calledWith(['Test Tag']);
+        });
+    });
+
+    it('assigns nothing when no tags match against multiple catalogs', function() {
+        var job = new GenerateTag({}, { target: 'bc7dab7e8fb7d6abf8e7d6ab' }, uuid.v4());
+        var tag = {
+            id: '1',
+            createdAt: new Date('Feb 01 2015 12:00:00'),
+            name: 'Test Tag',
+            rules: [
+                {
+                    path: 'dmi.dmi.system.product_name',
+                    equals: '1A'
+                },
+                {
+                    path: 'bmc.IP Address',
+                    equals: '192.168.1.1'
+                }
+            ]
+        };
+        waterline.tags.find.resolves([tag]);
+        waterline.catalogs.findMostRecent.onCall(0).resolves(catalog1);
+        waterline.catalogs.findMostRecent.onCall(1).resolves(catalog2);
+
+        return job.run()
+        .then(function() {
+            expect(waterline.nodes.addTags).to.have.not.been.called;
+        });
+    });
+
+
+    it('assigns nothing when no catalogs present', function() {
```

Comment:
Since you are doing this type of test, might be good to throw in an 'assigns nothing when no tags present' test as well?